interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -93,6 +93,8 @@
 
   public static final Metadata.Key<Messages.SimpleContext> METADATA_KEY =
       ProtoUtils.keyForProto(Messages.SimpleContext.getDefaultInstance());
+  private static final AtomicReference<Metadata.Headers> requestHeadersCapture =
```

Comment:
completely agreed. It's static because starting the server is static, and it goes into a `ServerInterceptor`. refactoring this test to use a new server for each test case is something I'd be happy to tackle, but it's outside the scope of this PR :)